corehq/apps/userreports/expressions/specs.py

```diff
@@ -327,6 +328,36 @@ def _get_subcases(self, case_id, context):
         return subcases
 
 
+class CaseSharingGroupsExpressionSpec(JsonObject):
+    type = TypeProperty('get_case_sharing_groups')
+    user_id_expression = DictProperty(required=True)
+
+    def configure(self, user_id_expression):
+        self._user_id_expression = user_id_expression
+
+    def __call__(self, item, context=None):
+        user_id = self._user_id_expression(item, context)
+        if not user_id:
+            return []
+
+        assert context.root_doc['domain']
+        return self._get_case_sharing_groups(user_id, context)
+
+    def _get_case_sharing_groups(self, user_id, context):
+        domain = context.root_doc['domain']
+        cache_key = (self.__class__.__name__, user_id)
```

Comment:
not really necessary but could include `domain` in the key as a safety for future in case the caching mechanism changes or something